cpe_workspaceone/libraries/hubcli.rb

```diff
@@ -70,17 +70,34 @@ def ws1_device_attributes
     def ws1_hubcli_exists
       @ws1_hubcli_exists ||=
         begin
-          hubcli_path = node['cpe_workspaceone']['hubcli_path']
-          ::File.exists?(hubcli_path)
+          !node['cpe_workspaceone']['hubcli_path'].nil? && ::File.exists?(node['cpe_workspaceone']['hubcli_path'])
         end
     end
 
-    def _get_available_ws1_profiles_list(hubcli_path)
+    def hubcli_path
+      return 'hubcli' unless ws1_hubcli_exists
+
+      return node['cpe_workspaceone']['hubcli_path'].gsub(/ /, '\ ')
+    end
+
+    def hubcli_execute(cmd, execute=true)
+      unless ws1_hubcli_exists
+        raise "Tried to execute hubcli, hubcli does not exist"
+      end
+
+      cmd = "#{hubcli_path} #{cmd.strip}"
+      if execute
+        cmd = shell_out(cmd)
+      end
+      cmd
+    end
+
+    def _get_available_ws1_profiles_list
       attributes = {}
       if node.macos?
         # spaces in path, so we need to convert them with gsub
-        cmd = shell_out(
-          "#{hubcli_path.gsub(/ /, '\ ')} profiles --list --json",
+        cmd = hubcli_execute(
+          "profiles --list --json",
```

Comment:
Older versions of Intelligent Hub do not support `profiles`, but this command silently fails. Making it fail loudly, or otherwise checking version, might be wise.

Reply:
Does it matter? Shouldn't it return an empty array? Perhaps we could check the info.plist off the agent that contains hubcli, but I wouldn't want multiple calls to hubcli just to understand the version.

Reply:
It doesn't really, and my iteration I didn't change it. I rather agree with you, but wanted to flag it just in case!

Reply:
I think we should return on any profile operation, warn and return if its an old version of hubcli. Checking the plist or shelling out doesn't matter much imo (both are going to be a new process)